### DISCUSSION TITLE
Starting  pypr on OpenSuse Tumbleweed

### DISCUSSION THREAD

**Discussion by btesteracc** [hyprland-community/pyprland#137] [General]
Hello,
I just started to use Hyprland and pypr. I'm on OpenSuse tumbleweed. pypr is started from /usr/local/bin/pypr, which is in the search path.
```
└─[$] <> bat /usr/local/bin/pypr 
───────┬──────────────────────────────────────────
       │ File: /usr/local/bin/pypr
───────┼──────────────────────────────────────────
   1   │ #!/bin/sh
   2   │ export PYTHONPATH='/var/cache/pypr/package.whl'
   3   │ exec python -m pyprland.command $@
───────┴──────────────────────────────────────────
```
I get an error that python was not found. On tumbleweed you start python with python3. Therefore I changed the line 3 from python to python3. Not sure if this works also for other distros.

**Comment by gitzalotl** [comment]
The solution suggested by @btesteracc worked for me, running openSUSE Tumbleweed.

In my case... 
Before the fix, a SyntaxError was thrown:
```
$ pypr
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 163, in _run_module_as_main
    mod_name, _Error)
  File "/usr/lib64/python2.7/runpy.py", line 119, in _get_module_details
    code = loader.get_code(mod_name)
  File "/var/cache/pypr/package.whl/pyprland/command.py", line 21
    CONTROL = f"{IPC_FOLDER}/.pyprland.sock"
                                           ^
SyntaxError: invalid syntax
```

**The error was thrown because Python 2.7 was being used to run code that requires Python 3.6 or higher.** 
The syntax error is valid because Python 2 does not recognize f-strings (formatted string literals), which were introduced in Python 3.6 - if I'm not mistaken.

Side note: `pip list` and `pip3 list` both showed  `pyprland` installed.

Let's see what's in the file that `pypr` was running...
```
$ which pypr    
/usr/local/bin/pypr

$ bat /usr/local/bin/pypr
───────┬──────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /usr/local/bin/pypr
───────┼──────────────────────────────────────────────────────────────────────────────────────────────
   1   │ #!/bin/sh
   2   │ export PYTHONPATH='/var/cache/pypr/package.whl'
   3   │ exec python -m pyprland.command $@
───────┴──────────────────────────────────────────────────────────────────────────────────────────────

$ which python
/usr/bin/python

$ python --version 
Python 2.7.18
```

Yep, python 2 was being used. Let's check python3.

```
$ which python3
/usr/bin/python3

$ python3 --version
Python 3.11.11
```

Okay. Let's apply the fix by running the code using python3.
For example, this fix can be applied by using a text editor (like `nano`, `vi`, `vim`, `nvim`) or by using `sed` to modify the file associated with command `pypr`. Below I show how `sed` can be used to change the content of this file in-place.

```
$ sudo sed -i 's/python/python3/' /usr/local/bin/pypr

$ bat /usr/local/bin/pypr
───────┬──────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /usr/local/bin/pypr
───────┼──────────────────────────────────────────────────────────────────────────────────────────────
   1   │ #!/bin/sh
   2   │ export PYTHONPATH='/var/cache/pypr/package.whl'
   3   │ exec python3 -m pyprland.command $@
───────┴──────────────────────────────────────────────────────────────────────────────────────────────

$ pypr
```
`pypr` runs OK now! :smile:

So, the fix was to modify the existing `/usr/local/bin/pypr` on my system to use `python3` instead of `python`.
